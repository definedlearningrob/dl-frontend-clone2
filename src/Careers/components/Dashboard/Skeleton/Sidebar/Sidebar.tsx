import { DCLogo } from '@dc/shared/DCLogo/DCLogo';

import SkeletonSidebarFooter from './Footer/Footer';
import SkeletonSidebarItem from './Item/Item';

type Props = {
  itemsCount: number;
};

function SkeletonSidebar({ itemsCount }: Props) {
  const renderItems = () =>
    Array.from({ length: itemsCount }, (_, index) => <SkeletonSidebarItem key={index} />);

  return (
    <div className='skeleton-sidebar'>
      <DCLogo className='skeleton-sidebar__logo-icon' />
      {renderItems()}
      <div className='skeleton-sidebar__divider' />
      <SkeletonSidebarFooter />
    </div>
  );
}

export default SkeletonSidebar;
