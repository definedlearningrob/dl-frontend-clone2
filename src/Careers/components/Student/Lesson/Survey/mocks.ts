import lessonInCourseQuery from '@dc/graphql/student/queries/lessonInCourse';
import { CREATE_CAREER_REVIEW_SURVEY_ATTEMPT } from '@dc/graphql/student/mutations/createCareerReviewSurveyAttempt';
import createCareerReviewSurveyAnswersMutation from '@dc/graphql/student/mutations/createCareerReviewSurveyAnswers';
import currentCoursesQuery from '@dc/graphql/student/queries/currentCourses';
import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

export const createAttemptSpy = jest.fn();
export const lessonInCourseSpy = jest.fn();

export let createCareerReviewSurveyAnswersSpy = jest.fn();
export let currentCoursesSpy = jest.fn();

export const mocks = [
  {
    request: {
      query: lessonInCourseQuery,
      variables: { courseId: '2', lessonId: '111', track: true },
    },
    result: () => {
      lessonInCourseSpy();

      return {
        data: {
          course: {
            careerName: 'Dental Laboratory Technicians',
            content: {
              checkIns: [],
              items: [],
            },
            hasInstitutionsInStudentState: true,
            id: '2',
            lesson: {
              assignments: [],
              attachments: [],
              careerReviewSurvey: {
                version: 1,
                performed: false,
                questions: [
                  {
                    answer: [],
                    id: '1',
                    options: [
                      {
                        option: 'Yes, definitely',
                        step: 1,
                      },
                      {
                        option: 'Definitely not',
                        step: 2,
                      },
                    ],
                    question: 'Can blind people see their dreams?',
                    type: 'single_choice' as SURVEY_QUESTION_TYPE,
                  },
                  {
                    answer: [],
                    id: '2',
                    options: [
                      {
                        option: 'Very Interested',
                        step: 1,
                      },
                      {
                        option: 'Not at all Interested',
                        step: 2,
                      },
                      {
                        option: 'Other',
                        step: 3,
                      },
                    ],
                    question: 'How interested are you in the Pathway this course is a part of?',
                    type: 'multiple_choice' as SURVEY_QUESTION_TYPE,
                  },
                ],
                __typename: 'CareerReviewSurvey',
              },
              checkInGroups: [],
              checkInQuestions: [],
              description: {
                introduction: null,
                goal: null,
                role: null,
                audience: null,
                situation: null,
                __typename: 'LessonDescription',
              },
              hasPresentation: false,
              id: '609',
              imageUrl: 'http://test.com/image.png',
              name: 'Career Review Survey',
              externalPresentations: [],
              researchLinks: [],
              texts: [],
              type: 'career_review_survey',
              videos: [],
              vocabularies: [],
              __typename: 'Lesson',
            },
            lessons: [
              {
                id: '782',
                step: 4,
                careerReviewSurvey: null,
              },
              {
                id: '609',
                step: 5,
                careerReviewSurvey: {
                  performed: false,
                  version: 1,
                  __typename: 'CareerReviewSurvey',
                },
              },
              {
                id: '275',
                step: 1,
                careerReviewSurvey: null,
              },
              {
                id: '331',
                step: 2,
                careerReviewSurvey: null,
              },
              {
                id: '668',
                step: 3,
                careerReviewSurvey: null,
              },
            ],
            name: 'Dental Laboratory Technician',
            progress: {
              submitted: 10,
              total: 10,
            },
            reviewSurvey: {
              questions: [
                {
                  id: '1',
                },
                {
                  id: '2',
                },
                {
                  id: '3',
                },
                {
                  id: '4',
                },
                {
                  id: '5',
                },
                {
                  id: '6',
                },
                {
                  id: '7',
                },
                {
                  id: '8',
                },
                {
                  id: '9',
                },
                {
                  id: '10',
                },
                {
                  id: '11',
                },
              ],
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: CREATE_CAREER_REVIEW_SURVEY_ATTEMPT,
      variables: { input: { contextType: 'COURSE', contextId: '1' } },
    },
    result: () => {
      createAttemptSpy();

      return {
        data: {
          createCareerReviewSurveyAttempt: {
            careerReviewSurveyAttempt: {
              id: '2',
              status: 'IN_PROGRESS',
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: createCareerReviewSurveyAnswersMutation,
      variables: {
        input: {
          contextId: '1',
          contextType: 'COURSE',
          answers: [
            { questionId: '1', answer: ['Yes, definitely'] },
            { questionId: '2', answer: ['Very Interested'] },
          ],
        },
      },
    },
    result: () => {
      createCareerReviewSurveyAnswersSpy();

      return {
        data: {
          createCareerReviewSurveyAnswers: {
            status: 'success',
          },
        },
      };
    },
  },
  {
    request: {
      query: currentCoursesQuery,
    },
    result: () => {
      currentCoursesSpy();

      return {
        data: {
          currentCourses: {
            id: '1',
            imageUrl: '',
            name: '',
            progress: {
              submitted: 2,
              total: 5,
            },
            status: 'success',
            pathway: {
              name: '',
            },
            thumbnailUrl: '',
            type: 'survey',
          },
        },
      };
    },
  },
];
