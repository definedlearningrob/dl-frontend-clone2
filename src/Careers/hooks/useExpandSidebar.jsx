import { createContext, useContext, useState } from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';

import { ROLES } from '@dc/resources/constants';

ExpandSidebarProvider.propTypes = {
  children: PropTypes.node.isRequired,
  value: PropTypes.object,
};

const ExpandSidebarContext = createContext();

export function ExpandSidebarProvider({ children, value }) {
  const history = useHistory();

  const [hideHeader, setHideHeader] = useState(false);

  const adminDashboardObject = {
    value: ROLES.ENTITY_ADMIN,
    label: 'entityAdmin',
    dashboardLink: '/',
  };
  const teacherDashboardObject = {
    value: ROLES.TEACHER,
    label: 'teacher',
    dashboardLink: '/teacher-dashboard',
  };

  const getInitialDashboardState = () => {
    const adminTeacherDashboardInitialPathRegex = /teacher-dashboard$/;
    const { pathname } = history.location;

    return adminTeacherDashboardInitialPathRegex.test(pathname)
      ? { active: teacherDashboardObject, option: adminDashboardObject }
      : { active: adminDashboardObject, option: teacherDashboardObject };
  };

  const [{ active, option }, setDashboardType] = useState(getInitialDashboardState);

  const toggleDashboardType = () => {
    setDashboardType({
      active: option,
      option: active,
    });

    history.push(option.dashboardLink);
  };

  return (
    <ExpandSidebarContext.Provider
      value={{
        activeDashboard: active,
        hideHeader,
        optionDashboard: option,
        setHideHeader,
        toggleDashboardType,
        ...value,
      }}>
      {children}
    </ExpandSidebarContext.Provider>
  );
}

function useExpandSidebar() {
  const { activeDashboard, hideHeader, optionDashboard, setHideHeader, toggleDashboardType } =
    useContext(ExpandSidebarContext);

  return {
    activeDashboard,
    hideHeader,
    optionDashboard,
    setHideHeader,
    toggleDashboardType,
  };
}

export default useExpandSidebar;
