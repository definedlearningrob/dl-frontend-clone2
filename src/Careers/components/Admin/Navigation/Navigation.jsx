import cx from 'classnames';
import { useTranslation } from 'react-i18next';

import { ReactComponent as BookOutlined } from '@dc/svg/book_outlined.svg';
import { ReactComponent as Entity } from '@dc/svg/home_dashboard.svg';
import { ReactComponent as VIIcon } from '@dc/svg/match.svg';

import { ReactComponent as RubricIcon } from '@shared/assets/icons/table.svg';
import { ReactComponent as BadgeIcon } from '@shared/assets/icons/award.svg';
import { SidebarNavItem } from '@shared/components/Sidebar/NavItem/NavItem';
import { ReactComponent as User } from '@shared/svg/user_outlined.svg';
import { ReactComponent as SettingsOutlined } from '@shared/svg/settings_outlined.svg';
import { ReactComponent as SchoolClass } from '@shared/svg/class.svg';
import { ReactComponent as Student } from '@shared/svg/student.svg';
import { ReactComponent as Tag } from '@shared/svg/tag_icon.svg';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';

function AdminNavigation() {
  const { isExpanded } = useNavigation();

  const { t } = useTranslation();

  const mainNavClasses = cx('navbar', {
    '-small': !isExpanded,
    '-animation-open': !isExpanded,
    '-animation-close': isExpanded,
  });

  const dividerClasses = cx('admin-nav-divider', { '-small': !isExpanded });

  const divider = (groupName) => (
    <div className={dividerClasses}>
      {groupName && <span className='admin-nav-divider__text'>{groupName}</span>}
      <hr className='admin-nav-divider__line' />
    </div>
  );

  const entitiesNavItems = [
    {
      id: 'entities',
      icon: <Entity />,
      path: '/admin/entities',
      text: t('sidebar.adminNavigation.entities'),
    },
    {
      id: 'users',
      icon: <User />,
      path: '/admin/users',
      text: t('sidebar.adminNavigation.users'),
    },
    {
      id: 'schoolclasses',
      icon: <SchoolClass />,
      path: '/admin/school-classes',
      text: t('sidebar.adminNavigation.schoolClasses'),
    },
    {
      id: 'students',
      icon: <Student />,
      path: '/admin/students',
      text: t('sidebar.adminNavigation.students'),
    },
    {
      id: 'virtualInternships',
      icon: <VIIcon />,
      path: '/admin/virtual-internships',
      text: t('sidebar.adminNavigation.virtualInternships'),
    },
  ].map((item) => <SidebarNavItem key={item.id} {...item} />);

  const courseNavItems = [
    {
      id: 'courses',
      icon: <BookOutlined />,
      path: '/admin/courses',
      text: t('sidebar.adminNavigation.courses'),
    },
    {
      id: 'lessons',
      icon: <BookOutlined />,
      path: '/admin/lessons',
      text: t('sidebar.adminNavigation.lessons'),
    },
    {
      id: 'lesson-items',
      icon: <BookOutlined />,
      path: '/admin/lesson-items',
      text: t('sidebar.adminNavigation.lessonItems'),
    },
  ].map((item) => <SidebarNavItem key={item.id} {...item} />);

  const planNavItems = [
    {
      id: 'plans',
      icon: <BookOutlined />,
      path: '/admin/plans',
      text: t('sidebar.adminNavigation.plan'),
    },
    {
      id: 'plan-groups',
      icon: <BookOutlined />,
      path: '/admin/plan-groups',
      text: t('sidebar.adminNavigation.planGroups'),
    },
    {
      id: 'standard-sets',
      icon: <BookOutlined />,
      path: '/admin/standard-sets',
      text: t('sidebar.adminNavigation.standardSets'),
    },
  ].map((item) => <SidebarNavItem key={item.id} {...item} />);

  const tagsNavItem = (
    <SidebarNavItem
      icon={<Tag />}
      id='tags'
      path='/admin/performance-indicators'
      text={t('sidebar.adminNavigation.performanceIndicators')}
    />
  );

  const contractsNavItem = (
    <SidebarNavItem
      {...{
        id: 'contracts',
        icon: <BookOutlined />,
        path: '/admin/contracts',
        text: t('sidebar.adminNavigation.contracts'),
      }}
    />
  );

  const checkInItems = [
    {
      id: 'check-ins',
      icon: <BookOutlined />,
      path: '/admin/check-ins',
      text: t('sidebar.adminNavigation.checkIns'),
    },
    {
      id: 'check-in-items',
      icon: <BookOutlined />,
      path: '/admin/checkin-groups',
      text: t('sidebar.adminNavigation.checkinGroups'),
    },
  ].map((item) => <SidebarNavItem key={item.id} {...item} />);

  const rubricsNavItem = (
    <SidebarNavItem
      {...{
        id: 'rubrics',
        icon: <RubricIcon />,
        path: '/admin/rubrics',
        text: t('sidebar.adminNavigation.rubrics'),
      }}
    />
  );
  const badgesNavItem = (
    <SidebarNavItem
      {...{
        id: 'badges',
        icon: <BadgeIcon />,
        path: '/admin/badges',
        text: t('sidebar.adminNavigation.badges'),
      }}
    />
  );

  const catalogNavItems = [
    {
      id: 'tasks',
      icon: <BookOutlined />,
      path: '/admin/tasks',
      text: t('sidebar.adminNavigation.tasks'),
    },
    {
      id: 'products',
      icon: <BookOutlined />,
      path: '/admin/products',
      text: t('sidebar.adminNavigation.products'),
    },
  ].map((item) => <SidebarNavItem key={item.id} {...item} />);

  const sharedCatalogs = [
    {
      id: 'catalogs',
      icon: <BookOutlined />,
      path: '/admin/catalogs',
      text: t('sidebar.adminNavigation.catalogs'),
    },
    {
      id: 'tracks',
      icon: <BookOutlined />,
      path: '/admin/tracks',
      text: t('sidebar.adminNavigation.tracks'),
    },
    {
      id: 'units',
      icon: <BookOutlined />,
      path: '/admin/units',
      text: t('sidebar.adminNavigation.units'),
    },
  ].map((item) => <SidebarNavItem key={item.id} {...item} />);

  const maintenanceNavItem = (
    <SidebarNavItem
      key='maintenance'
      {...{
        id: 'maintenance',
        icon: <SettingsOutlined />,
        path: '/admin/maintenance',
        text: t('sidebar.adminNavigation.maintenance'),
      }}
    />
  );

  return (
    <nav className={mainNavClasses}>
      {divider(t('sidebar.adminNavigation.dc'))}
      {entitiesNavItems}
      {courseNavItems}
      {planNavItems}
      {tagsNavItem}
      {contractsNavItem}
      {checkInItems}
      {rubricsNavItem}
      {badgesNavItem}
      {divider(t('sidebar.adminNavigation.pbl'))}
      {catalogNavItems}
      {divider(t('sidebar.adminNavigation.shared'))}
      {sharedCatalogs}
      {divider()}
      {maintenanceNavItem}
    </nav>
  );
}

export default AdminNavigation;
