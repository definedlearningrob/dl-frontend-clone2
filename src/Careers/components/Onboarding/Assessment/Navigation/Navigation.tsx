import { useState } from 'react';
import { useTranslation } from 'react-i18next';

import QuitModal from '@dc/components/Onboarding/Assessment/Navigation/QuitModal/QuitModal';
import useAssessment from '@dc/hooks/useAssessment';
import { ASSESSMENT_TYPES } from '@dc/resources/constants';
import useUserInfo from '@dc/hooks/useUserInfo';
import { ReactComponent as LogoType } from '@dc/svg/logotype.svg';
import { TStudentInfo } from '@dc/graphql/student/queries/userInfo';

import SharedButton from '@shared/components/Button/Button';
import SharedIcon from '@shared/components/Icon/Icon';
import { LanguageSwitcher } from '@shared/components/LanguageSwitcher/LanguageSwitcher';
import { useLocalize } from '@shared/hooks/useLocalize';
import { ReadSpeaker } from '@shared/components/ReadSpeaker/ReadSpeaker';

type Props = {
  isSaving?: boolean;
};

function OnboardingAssessmentNavigation({ isSaving }: Props) {
  const { t } = useTranslation();
  const { localesToSelect, selectedLocale, setLanguage } = useLocalize();
  const {
    assessment: {
      attempt: { assessmentType },
    },
  } = useAssessment();
  const {
    userInfo: {
      hasCompletedOnboarding,
      isImpersonated,
      firstName,
      lastName,
      settings: { onboardingEnabled },
    },
  } = useUserInfo<TStudentInfo>();
  const [isModalOpen, setIsModalOpen] = useState(false);

  const toggleModal = () => setIsModalOpen(!isModalOpen);

  const renderActions = () => {
    if (assessmentType === ASSESSMENT_TYPES.HIGH_SCHOOL) {
      return hasCompletedOnboarding || !onboardingEnabled
        ? t('student.onboarding.assessment.abort.backToDashboard')
        : t('student.onboarding.assessment.abort.logout');
    }

    if (assessmentType === ASSESSMENT_TYPES.MIDDLE_SCHOOL) {
      return t('student.onboarding.assessment.abort.saveProgress');
    }
  };

  return (
    <>
      {isImpersonated && (
        <div className='assessment__impersonating-header'>
          {t('student.onboarding.assessment.impersonating', {
            name: `${firstName} ${lastName}`,
          })}
          <button className='assessment__impersonating-button' onClick={toggleModal}>
            {t('appHeader.exitSession')}
          </button>
        </div>
      )}
      <header className='assessment__navbar'>
        <SharedButton isLoading={isSaving} variant='primary-outlined' onClick={toggleModal}>
          <>{renderActions()}</>
        </SharedButton>
        <div className='flex gap-sm items-center'>
          <SharedIcon className='h-base border-r border-neutral-300 pr-sm' icon={<LogoType />} />
          <p className='m-0 font-bold text-xs'>{t('student.onboarding.assessment.title')}</p>
        </div>
        <div className='[&>*]:mb-0 [&_a>span]:!pl-xs'>
          <ReadSpeaker />
        </div>
        <div>
          <LanguageSwitcher
            languages={localesToSelect}
            selectedLanguage={selectedLocale}
            setLanguage={setLanguage}
          />
        </div>
      </header>
      <QuitModal
        isFirstAssessment={!hasCompletedOnboarding && onboardingEnabled}
        isOpen={isModalOpen}
        onClose={toggleModal}
      />
    </>
  );
}

export default OnboardingAssessmentNavigation;
