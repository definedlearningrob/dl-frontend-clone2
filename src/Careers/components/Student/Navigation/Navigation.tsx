import { useTranslation } from 'react-i18next';
import { useOverallProgressQuery } from '@graphql/dc/students/hooks';

import useUserInfo from '@dc/hooks/useUserInfo';
import { ReactComponent as CoursesIcon } from '@dc/svg/courses.svg';
import { ReactComponent as DashboardIcon } from '@dc/svg/home_dashboard.svg';
import { ReactComponent as FinalReportIcon } from '@dc/svg/file_document.svg';
import { ReactComponent as MessagesIcon } from '@dc/svg/messages.svg';
import { ReactComponent as OpportunitiesIcon } from '@dc/svg/match.svg';
import { ReactComponent as PostSecondaryIcon } from '@dc/svg/test-checkmark-done.svg';
import { ReactComponent as ExperiencesSummaryIcon } from '@dc/svg/experiencesSummary.svg';
import { TStudentInfo } from '@dc/graphql/student/queries/userInfo';

import { ReactComponent as PlansIcon } from '@shared/svg/certificate.svg';
import { ReactComponent as PortfolioIcon } from '@shared/svg/portfolio.svg';
import { SidebarNavItem } from '@shared/components/Sidebar/NavItem/NavItem';
import { useFeatureFlags } from '@shared/components/FeatureProvider';
import { QuestionBox } from '@shared/components/QuestionBox/QuestionBox';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';

import styles from './Navigation.module.sass';

export const StudentNavigation = () => {
  const { t } = useTranslation();
  const { isExpanded } = useNavigation();
  const { userInfo } = useUserInfo<TStudentInfo>();
  const { POST_SECONDARY_ON } = useFeatureFlags();
  const { data: overallProgressData } = useOverallProgressQuery();
  const isOpportunitiesFeatureDisabled = !userInfo.hasOpportunitiesEnabled;
  const isPostSecondaryFeatureDisabled = !POST_SECONDARY_ON;
  const isEnrolledInCourse = overallProgressData?.overallProgress?.enrolledInCourse;

  const navItems = [
    {
      id: 'dashboard',
      icon: <DashboardIcon />,
      path: '/',
      text: t('student.navigation.dashboard'),
      childrenItems: [
        {
          id: 'final-report',
          icon: <FinalReportIcon />,
          path: '/final-report',
          hidden: !userInfo?.hasCompletedAssessment && !isEnrolledInCourse,
          text: t('student.navigation.finalReport'),
        },
        {
          id: 'final-report',
          icon: <ExperiencesSummaryIcon />,
          path: '/portfolio/experiences',
          hidden: !userInfo?.hasCompletedAssessment,
          text: t('student.navigation.experiencesSummary'),
        },
      ],
    },
    {
      id: 'courses',
      icon: <CoursesIcon />,
      path: '/courses',
      text: t('student.navigation.courses'),
    },
    {
      id: 'opportunities',
      icon: <OpportunitiesIcon />,
      hidden: isOpportunitiesFeatureDisabled,
      path: '/opportunities',
      text: t('student.navigation.opportunities'),
    },
    {
      id: 'post-secondary',
      icon: <PostSecondaryIcon />,
      path: '/post-secondary',
      text: t('student.navigation.postSecondary'),
      hidden: isPostSecondaryFeatureDisabled,
    },
    {
      id: 'portfolio',
      icon: <PortfolioIcon />,
      path: '/portfolio',
      text: t('student.navigation.portfolio'),
    },
    {
      id: 'plans',
      icon: <PlansIcon />,
      path: '/plans',
      text: t('student.navigation.plans'),
      hidden: !userInfo.hasPlans,
    },
    {
      id: 'messages',
      icon: <MessagesIcon />,
      path: '/messages',
      needsAttention: userInfo?.hasUnreadConversation,
      text: t('student.navigation.messages'),
    },
  ];

  const renderNavItems = navItems.map((item) => <SidebarNavItem key={item.id} {...item} />);

  return (
    <>
      <nav className={styles.navigation}>{renderNavItems}</nav>
      <QuestionBox isExpanded={isExpanded} />
    </>
  );
};
