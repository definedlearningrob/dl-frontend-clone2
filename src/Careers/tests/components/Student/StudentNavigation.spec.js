import { createMemoryHistory } from 'history';
import { fireEvent, screen, waitFor } from '@testing-library/dom';
import { I18nextProvider } from 'react-i18next';
import { MockedProvider } from '@apollo/client/testing';

import i18n from '@dc/i18n';
import { StudentNavigation } from '@dc/components/Student/Navigation/Navigation';
import { ExpandSidebarProvider } from '@dc/hooks/useExpandSidebar';
import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import { studentInfoMock } from '@dc/tests/mocks/studentMocks';

import { Sidebar } from '@shared/components/Sidebar/Sidebar';
import { MessagingProvider } from '@shared/hooks/useMessaging';
import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';
const history = createMemoryHistory();

history.replace = jest.fn();

const getSideBarItem = (number) =>
  screen.getAllByTestId(/navigation-item/)[number].querySelector('span');

const renderSideBar = () =>
  renderWithRouterAndReduxProvider(
    <MockedProvider mocks={[studentInfoMock]}>
      <I18nextProvider i18n={i18n}>
        <UserInfoProvider value={{ userInfo: studentInfoMock.result.data.userInfo }}>
          <ExpandSidebarProvider>
            <NavigationContextProvider>
              <MessagingProvider refreshUser={jest.fn()} userInfo={{}}>
                <Sidebar LogoComponent={() => <div>Logo</div>}>
                  <StudentNavigation />
                </Sidebar>
              </MessagingProvider>
            </NavigationContextProvider>
          </ExpandSidebarProvider>
        </UserInfoProvider>
      </I18nextProvider>
    </MockedProvider>,
    { history, initialState: { session: { user: { type: 'student' } } } }
  );

describe('StudentNavigation', () => {
  it('renders correctly', async () => {
    const { container } = renderSideBar();

    expect(container).toMatchSnapshot();
  });

  it('shows all options with visible text', () => {
    renderSideBar();

    const firstItem = getSideBarItem(0);
    const secondItem = getSideBarItem(1);

    expect(firstItem).toHaveTextContent('Dashboard');
    expect(firstItem).toHaveClass('text');

    expect(secondItem).toHaveTextContent('Courses');
    expect(secondItem).toHaveClass('text');
  });

  it('navigates properly', async () => {
    renderSideBar();

    const dashboardRoute = screen.getByTestId(/navigation-item-dashboard/);
    const coursesRoute = screen.getByTestId(/navigation-item-courses/);

    fireEvent.click(dashboardRoute);
    await waitFor(() => expect(history.location.pathname).toBe('/'));

    fireEvent.click(coursesRoute);
    await waitFor(() => expect(history.location.pathname).toBe('/courses'));
  });

  it('switches visibility on hide/show button click', async () => {
    renderSideBar();
    const firstItem = getSideBarItem(0);
    const secondItem = getSideBarItem(1);

    const changeSizeButton = screen.getByTestId(/sidebar-expand-button/);

    fireEvent.click(changeSizeButton);

    expect(firstItem).toHaveClass('text');
    expect(firstItem).not.toHaveClass('opacity-0');

    expect(secondItem).toHaveClass('text');
    expect(secondItem).not.toHaveClass('opacity-0');

    await waitFor(() => fireEvent.click(changeSizeButton));

    await waitFor(() => {
      expect(firstItem).toHaveClass('font-medium truncate text opacity-0');
      expect(secondItem).toHaveClass('font-medium truncate text opacity-0');
    });
  });
});
