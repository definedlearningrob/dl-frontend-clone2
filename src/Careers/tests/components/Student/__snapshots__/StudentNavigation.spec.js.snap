// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentNavigation renders correctly 1`] = `
<div>
  <div
    class="sidebar flex flex-col fixed top-0 z-higher bg-white shadow-200 h-screen animationOpen small"
  >
    <div>
      <div>
        Logo
      </div>
    </div>
    <nav
      class="navigation"
    >
      <div
        class="relative"
      >
        <span
          data-state="closed"
        >
          <a
            aria-current="page"
            class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item hasChildren text-primary-500 bg-primary-200"
            data-testid="navigation-item-dashboard"
            href="/"
          >
            <div
              class="icon -xs-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="font-medium truncate text opacity-0 me-auto"
            >
              Dashboard
            </span>
          </a>
        </span>
        <div
          class="transition-[right] duration-300 ease-in-out absolute top-sm right-[-20px]"
        >
          <div
            aria-expanded="false"
            aria-haspopup="menu"
            class="relative right-xs"
            data-state="closed"
            id="radix-0"
            type="button"
          >
            <span
              data-state="closed"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0 bg-white text-primary-500 rounded-full outline outline-transparent transition-[outline-color] ease-linear duration-200 p-xxs"
                  data-testid="icon"
                >
                  <div
                    class="h-sm w-sm"
                  />
                </div>
              </div>
            </span>
          </div>
        </div>
      </div>
      <div
        class="relative"
      >
        <span
          data-state="closed"
        >
          <a
            class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
            data-testid="navigation-item-courses"
            href="/courses"
          >
            <div
              class="icon -xs-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="font-medium truncate text opacity-0"
            >
              Courses
            </span>
          </a>
        </span>
      </div>
      <div
        class="relative"
      >
        <span
          data-state="closed"
        >
          <a
            class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
            data-testid="navigation-item-opportunities"
            href="/opportunities"
          >
            <div
              class="icon -xs-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="font-medium truncate text opacity-0"
            >
              Opportunities
            </span>
          </a>
        </span>
      </div>
      <div
        class="relative"
      >
        <span
          data-state="closed"
        >
          <a
            class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
            data-testid="navigation-item-post-secondary"
            href="/post-secondary"
          >
            <div
              class="icon -xs-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="font-medium truncate text opacity-0"
            >
              College & Future
            </span>
          </a>
        </span>
      </div>
      <div
        class="relative"
      >
        <span
          data-state="closed"
        >
          <a
            class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
            data-testid="navigation-item-portfolio"
            href="/portfolio"
          >
            <div
              class="icon -xs-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="font-medium truncate text opacity-0"
            >
              Portfolio
            </span>
          </a>
        </span>
      </div>
      <div
        class="relative"
      >
        <span
          data-state="closed"
        >
          <a
            class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
            data-testid="navigation-item-plans"
            href="/plans"
          >
            <div
              class="icon -xs-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="font-medium truncate text opacity-0"
            >
              Plans
            </span>
          </a>
        </span>
      </div>
      <div
        class="relative"
      >
        <span
          data-state="closed"
        >
          <a
            class="flex items-center gap-xs pr-xs py-0 pl-sm text-xs text-neutral-700 no-underline pointer relative item"
            data-testid="navigation-item-messages"
            href="/messages"
          >
            <div
              class="icon -xs-size"
              data-testid="icon"
            >
              <div />
            </div>
            <span
              class="font-medium truncate text opacity-0"
            >
              Messages
            </span>
          </a>
        </span>
      </div>
    </nav>
    <div
      class="overflow-x-hidden pt-md"
    >
      <div
        class="animate-fadeDropIn w-[152px] bg-neutral-200 rounded-sm text-center p-xs mt-auto mx-xs mb-sm relative hidden"
      >
        <div
          class="flex justify-center -left-[10px] absolute -top-[20px] w-full"
        >
          <div
            class="icon h-[30px] w-[66px]"
            data-testid="icon"
          >
            <div />
          </div>
        </div>
        <p
          class="mb-xxs text-xxs tracking-[-0.24px] font-regular"
        >
          Ask teacher for help
        </p>
        <button
          class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] !text-xxs mx-auto -primary-outlined text-primary-500 hover:bg-primary-200 py-xs px-sm !text-xs leading-[14px] !gap-xxs border"
          data-testid="button"
          type="button"
        >
          <span
            class="items-center flex justify-center gap-xs"
          >
            Send message
          </span>
        </button>
      </div>
      <span
        data-state="closed"
      >
        <button
          class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] questionButton -default bg-white text-font-primary hover:bg-primary-200 outline-primary-500 border-white hover:border-primary-200 py-xs px-sm !text-xs leading-[14px] !gap-xxs border"
          data-testid="button"
          type="button"
        >
          <span
            class="items-center flex justify-center gap-xs"
          >
            <div
              class="icon -xs-size"
              data-testid="icon"
            >
              <div />
            </div>
          </span>
        </button>
      </span>
    </div>
    <div
      class="px-xs z-highest mb-sm collapsed"
      id="readspeaker_button1"
    >
      <div
        class="rs_skip rsbtn rs_preserve read-speaker flex items-center !mb-0 !bg-neutral-200 !rounded-sm !p-xxs"
      >
        <a
          accesskey="L"
          class="rsbtn_play !flex items-center grow !bg-transparent !border-r-0 !border-t-0 !border-b-0 !rounded-none !h-base !border-l-0"
          rel="nofollow"
          title="Listen to this page using Read Speaker web Reader"
        >
          <span
            class="rsbtn_left rsimg rspart grow block !m-0 !bg-neutral-200 hover:!bg-neutral-700 rounded-sm group !py-xxs"
          >
            <span
              class="rsbtn_text before:!content-none after:!content-none !flex items-center"
            >
              <span
                class="!p-0 grow !flex items-center justify-center gap-xxxs group-hover:!text-white"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0 text-primary-500 grow group-hover:!text-inherit !p-xxs"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
              </span>
            </span>
          </span>
        </a>
      </div>
    </div>
    <div
      class="mt-auto"
    >
      <div
        class="divider"
      />
      <span
        data-state="closed"
      >
        <div
          class="expandButton animationOpen"
          data-testid="sidebar-expand-button"
        >
          <div
            class="icon icon smallIcon -xs-size"
            data-testid="icon"
          >
            <div />
          </div>
          <span
            class="buttonText hidden"
          >
            Collapse
          </span>
        </div>
      </span>
    </div>
  </div>
</div>
`;
